### PR TITLE
apply DuplicateFreeList to output of CheckConstructivenessOfCategory

### PR DESCRIPTION
**Pull request by mohamed-barakat**

### CODE REVIEW

File: CAP/gap/CAP.gi [hunk 1]
```diff
@@ -618,7 +618,7 @@ InstallMethod( CheckConstructivenessOfCategory,
       
     od;
     
-    return result_list;
+    return DuplicateFreeList( result_list );
```
[the remark this answers is not in the document]
Comment (mohamed-barakat) [reply CAP/gap/CAP.gi:621]: Yes, the order comes from the stack of constructive properties, so I would like to keep the most relevant ones first.

**Comment by zickgraf** [comment]
I first thought the duplicate entries would come from looping over multiple properties, but now I see that already single entries of `CAP_INTERNAL_CONSTRUCTIVE_CATEGORIES_RECORD` contain duplicates because they are obtained by a concatenation of non-disjoint lists. So I would say deduplication should happen when creating those lists in the first place.

**Comment by mohamed-barakat** [comment]
I agree.

**Comment by mohamed-barakat** [comment]
I changed my mind. I think it is in any way reasonable to return a duplicate free list at this late stage.

**Comment by mohamed-barakat** [comment]
Closing this PR after a discussion with @zickgraf about a more elaborate interface to add categorical properties/doctrines that takes care of the implications.

**Pull request closed without merging**